src/chttpd_handlers.erl

```diff
@@ -51,6 +54,20 @@ design_handler(HandlerKey, DefaultFun) ->
 %% Internal Function Definitions
 %% ------------------------------------------------------------------
 
+is_disabled(KeyBin) ->
+    BlacklistConfig = config:get("chttpd", "url_handler_blacklist", []),
+    {ok, Blacklist} = couch_util:parse_term(BlacklistConfig),
+    Key = binary_to_list(KeyBin),
+    lists:member(Key, Blacklist).
```

Comment:
A (very) dirty version of this, just to illustrate the idea:

```erlang
get_blacklist() ->
    case config:get("chttpd", "_url_handler_blacklist", []) of
        [] -> % no cached version
            case config:get("chttpd", "url_handler_blacklist", []) of
                [] -> []; % no config value
                BlacklistConfig -> % we have a config value, let’s parse it once
                    {ok, Blacklist} = couch_util:parse_term(BlacklistConfig),
                    config:set("chttpd", "_url_handler_blacklist", Blacklist),
                    Blacklist
            end;
        Blacklist1 -> Blacklist1 % we already have the parsed list cached
    end.

is_disabled(KeyBin) ->
    Blacklist = get_blacklist(),
    Key = ?b2l(KeyBin),
    lists:member(Key, Blacklist).
```

Of course, we’d need a thing that invalidates the “cache”.

I’m abusing the config system here as a cache, we obviously don’t want to do that for real. I’m not too familiar with chttpd yet, would there be an obvious location for this?